src/index.tsx

```diff
@@ -210,7 +210,9 @@ export default function AxeMode({ children, disabled }: AxeModeProps) {
   const childrenRef = React.useRef<HTMLElement | null>(null);
 
   React.useEffect(() => {
-    import('axe-core').then(setAxe);
```

Comment:
I think it's probably OK to import the module regardless of whether or not it's disabled. You only want to do it once anyway, and that way if the user toggles from disabled after the component is mounted it should be ready to go (not that it would be that slow anyway, but still).

So I'd probably just check to make sure the component is still mounted in the async callback and leave out the disabled check.

```js
  React.useEffect(() => {
    let mounted = true;
    import('axe-core').then((axe) => {
      if (mounted) {
        setAxe(axe);
      }
    });
    return () => {
      mounted = false;
    }
  }, []);
```

Reply:
Hmm. I think we should check for `disabled` before importing the module.

Webpack will split `axe-mode` into a separate chunk since it is imported dynamically. Although, if `AxeMode` is wrapped around the application at all times, we should make sure not to import the dynamic module when it's disabled, which is done in user-land by setting `disabled={process.env.NODE_ENV !== 'development'`. Otherwise, we will always import the chunk and essentially this does not have the desired effect of not requesting it in production.

Reply:
It should probably be disabled completely in production anyway IMO. Maybe we're trying to optimize something a bit too early here, I'm not sure the bundle will be a bottleneck for development.

```js
const AxeModeImpl = () => () // real implementation
export const AxeMode = process.env.NODE_ENV !== 'development' ? React.Fragment : AxeModeImpl
```

Reply:
Totally agree!

I think in your example, `AxeMode` would have to be a component that imports `AxeModeImpl` based on `NODE_ENV`. Otherwise, it will be a top-level import and bundled alongside the application code.

I think we can import `axe-core` synchronously then?

Reply:
@raunofreiberg Yeah I think that's a somewhat better approach. I'll add to the PR.